crates/optimism/primitives/src/lib.rs

```diff
@@ -9,3 +9,172 @@
 
 pub mod bedrock;
 pub mod op_tx_type;
+
+use alloy_primitives::{bytes, Bytes, TxKind, Uint, B256};
+
+use alloy_consensus::{SignableTransaction, TxLegacy};
+use alloy_rlp::{Decodable, Encodable, Result};
+use op_tx_type::OpTxType;
+use reth_primitives::revm_primitives::{AccessList, SignedAuthorization};
+use reth_primitives_traits::{InMemorySize, TransactionExt};
+
+use op_alloy_consensus::{OpTxEnvelope, OpTypedTransaction};
+
+#[cfg(feature = "arbitrary")]
+use arbitrary::Arbitrary;
+
+#[cfg_attr(feature = "arbitrary", derive(Arbitrary))]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+/// Optimistic transaction type.
+pub struct OpTransaction(OpTypedTransaction);
+
+impl Default for OpTransaction {
+    fn default() -> Self {
+        Self(OpTypedTransaction::Legacy(TxLegacy::default()))
+    }
+}
+
+impl Encodable for OpTransaction {
+    fn encode(&self, out: &mut dyn bytes::BufMut) {
+        match &self.0 {
+            OpTypedTransaction::Legacy(tx) => tx.encode(out),
+            OpTypedTransaction::Eip2930(tx) => tx.encode(out),
+            OpTypedTransaction::Eip1559(tx) => tx.encode(out),
+            OpTypedTransaction::Eip7702(tx) => tx.encode(out),
+            OpTypedTransaction::Deposit(tx) => tx.encode(out),
+        }
+    }
+}
```

Comment:
sure the `RlpEncodable` derive macro doesn't work here? same goes for `RlpDecodable` instead of manual impl

Reply:
actually, I think the rlp impls can be removed completely, these aren't super traits of `Transaction` nor `TransactionExt`